src/main/java/com/cohere/api/core/Stream.java

```diff
@@ -3,95 +3,294 @@
      *
      * @param valueType The class of the objects in the stream.
      * @param reader    The reader that provides the streamed data.
      * @param delimiter The delimiter used to separate elements in the stream.
      */
     public Stream(Class<T> valueType, Reader reader, String delimiter) {
+        this.valueType = valueType;
         this.scanner = new Scanner(reader).useDelimiter(delimiter);
+        this.streamType = StreamType.JSON;
+        this.delimiter = delimiter;
+        this.uri = null;
+        this.sseReader = null;
+    }
+
+    /**
+     * Private constructor for SSE streaming with manual parsing.
+     *
+     * @param valueType The class of the objects in the stream.
+     * @param uri The URI of the SSE endpoint.
+     * @param sseReader The reader that provides the SSE data.
+     */
+    private Stream(Class<T> valueType, URI uri, Reader sseReader) {
         this.valueType = valueType;
+        this.scanner = null;
+        this.streamType = StreamType.SSE;
+        this.delimiter = null;
+        this.uri = uri;
+        this.sseReader = sseReader;
     }
 
     /**
-     * Returns an iterator over the elements in this stream that blocks during iteration when the next object is
-     * not yet available.
+     * Creates a new Stream for newline-delimited JSON format.
+     *
+     * @param <T> The type of objects in the stream.
+     * @param valueType The class of the objects in the stream.
+     * @param reader The reader that provides the JSON data.
+     * @param delimiter The delimiter used to separate JSON objects.
+     * @return A new Stream configured for JSON parsing.
+     */
+    public static <T> Stream<T> fromJson(Class<T> valueType, Reader reader, String delimiter) {
+        return new Stream<>(valueType, reader, delimiter);
+    }
+
+    /**
+     * Creates a new Stream for newline-delimited JSON format with default newline delimiter.
+     *
+     * @param <T> The type of objects in the stream.
+     * @param valueType The class of the objects in the stream.
+     * @param reader The reader that provides the JSON data.
+     * @return A new Stream configured for JSON parsing with newline delimiter.
+     */
+    public static <T> Stream<T> fromJson(Class<T> valueType, Reader reader) {
+        return new Stream<>(valueType, reader, "\\n");
+    }
+
+    /**
+     * Creates a new Stream for Server-Sent Events using manual parsing.
+     * This is designed to work with Cohere's SSE streaming endpoints.
+     *
+     * @param <T> The type of objects in the stream.
+     * @param valueType The class of the objects in the stream.
+     * @param uri The URI of the SSE endpoint.
+     * @param sseReader The reader that provides the SSE data.
+     * @return A new Stream configured for SSE streaming.
+     */
+    public static <T> Stream<T> fromSSE(Class<T> valueType, URI uri, Reader sseReader) {
+        return new Stream<>(valueType, uri, sseReader);
```

Comment:
https://github.com/humanloop/humanloop-node/blob/c8151f826a50a49a2536874df5b56d701092e28a/src/core/streaming-fetcher/Stream.ts#L28 @tstanmay13 i think we should take into account a `streamTerminator` which is how you know the stream is finished `[DONE]` is what it should default to if unspecified